bevy_lint/tests/ui/insert_event_resource/main.rs

```diff
@@ -0,0 +1,26 @@
+#![feature(register_tool)]
+#![register_tool(bevy)]
+#![deny(bevy::insert_event_resource)]
+
+use bevy::prelude::*;
+
+#[derive(Event)]
+struct Foo;
+
+fn main() {
+    App::new().init_resource::<Events<Foo>>();
+    //~^ ERROR: called `App::init_resource::<Events<T>>()` instead of `App::add_event::<T>()`
```

Comment:
For my understanding: Do the tests just check that this error is emitted for this line, or do they make an equality comparison of the CLI output with the `.stderr` file? Or both?

Reply:
To my understanding, it:

1. Generates and compares `.stderr` files.
2. Ensures that `//~` diagnostics are in those `.stderr` files, and that no diagnostics are missed.
3. Generates and compares `.fixed` files, if machine-applicable suggestions are made.
4. Ensures `.fixed` programs compile. (I don't know if it skips warnings or not.)
5. Generates and compares the executable in `target/ui`, to ensure the same binary is generated by `rustc`. (I don't think we need this, so I may try to disable it.)

It can be a bit confusing, since there's not much documentation on all of the features.